src/components/Transactions/transactions.js

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from 'react';
+import Box from '@mui/material/Box';
+import { useParams } from 'react-router-dom';
+import { getAccountQuery } from '../../services/Account/account';
+import { useAuth } from '../../utils/Auth/use-auth';
+import Navbar from '../Navbar/navbar';
+import { Card, CardContent, Typography, Button, Modal, Tooltip } from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import NotFound from '../NotFound/notFound';
+import './transactions.css';
+import { useHistory } from 'react-router-dom';
+import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos';
+import { currencyFormatter } from '../../utils/utils/format';
+import TransactionForm from '../TransactionForm/transactionForm';
+import { onSnapshot } from 'firebase/firestore';
+import { getTransactionsQuery } from '../../services/Transaction/transaction';
+import { Badge } from '@mui/material';
+import { TransactionType } from '../../utils/utils/constants';
+
+const Transactions = () => {
+    const auth = useAuth();
+    const { accountId } = useParams();
+    const [accountState, setAccountsState] = useState(undefined);
+    const [isLoading, setIsLoading] = useState(false);
+    const [open, setOpen] = useState(false);
+    const handleOpen = () => setOpen(true)
+    const handleClose = () => setOpen(false);
+    let history = useHistory();
+    const [transactionsState, setTransactionsState] = useState([]);
+
+    useEffect(() => {
+      if(auth.data && !auth.isLoading){
+        const transactionsQuery = getTransactionsQuery(accountId);
+        const unsubscribe = onSnapshot(transactionsQuery, (querySnapshot) => {
```

Comment:
Here you could move this part of the code into a custom hook to avoid repeating code making a custom hook that recieves the querySnapshot and returning data no matter from which collection you want to fetch